packages/components/src/slot-fill/context.js

```diff
@@ -21,22 +22,49 @@ class SlotFillProvider extends Component {
 	constructor() {
 		super( ...arguments );
 
-		this.registerSlot = this.registerSlot.bind( this );
-		this.registerFill = this.registerFill.bind( this );
-		this.unregisterSlot = this.unregisterSlot.bind( this );
-		this.unregisterFill = this.unregisterFill.bind( this );
-		this.getSlot = this.getSlot.bind( this );
-		this.getFills = this.getFills.bind( this );
-
 		this.slots = {};
 		this.fills = {};
 		this.state = {
-			registerSlot: this.registerSlot,
-			unregisterSlot: this.unregisterSlot,
-			registerFill: this.registerFill,
-			unregisterFill: this.unregisterFill,
-			getSlot: this.getSlot,
-			getFills: this.getFills,
+			registerSlot: this.proxy( 'registerSlot' ).bind( this ),
+			registerFill: this.proxy( 'registerFill' ).bind( this ),
+			unregisterSlot: this.proxy( 'unregisterSlot' ).bind( this ),
+			unregisterFill: this.proxy( 'unregisterFill' ).bind( this ),
+			getSlot: this.proxy( 'getSlot' ).bind( this ),
+			getFills: this.proxy( 'getFills' ).bind( this ),
+		};
+	}
+
+	/**
+	 * Given a function name for a function on the SlotFillProvider prototype,
+	 * returns a new function which either passes the arguments to the current
+	 * instance, or to the context ancestor, dependent on whether the provider
+	 * is configured to handle the given slot.
+	 *
+	 * @param {string} functionName SlotFillProvider function name.
+	 *
+	 * @return {Function} Proxying function.
+	 */
+	proxy( functionName ) {
+		return ( name, ...args ) => {
+			const { slots: handledSlots } = this.props;
+
+			let handler = this[ functionName ];
+
+			if ( Array.isArray( handledSlots ) ) {
+				const isHandled = some( handledSlots, ( slotNameOrComponent ) => {
+					const { slotName = slotNameOrComponent } = slotNameOrComponent;
```

Comment:
This works well and is smart but makes the code a little hard to follow. Maybe we can just assign the name with const `slotName = ( typeof slotNameOrComponent === 'string' ) slotNameOrComponent : slotNameOrComponent.slotName;`
More verbose but I guess it is easier to follow.

Reply:
> This works well and is smart but makes the code a little hard to follow.

Yeah, in retrospect I think you're right that I may have tried to be a little too clever here. Your simpler alternative seems to be an improvement.